polyply/src/map_to_molecule.py

```diff
@@ -30,131 +27,236 @@ def tag_exclusions(blocks, force_field):
+        to the blocks in the force-field. It does the essential
+        bookkeeping for three cases and populates the node_to_block,
+        and node_to_fragment dicts as well as the fragments attribute.
+        It distinguishes three cases:
+
+        1) the node corresponds to a single residue block; here
+           node_to_block entry is simply the resname of the block
+        2) the node has the from_itp attribute; in this case
+           the node is part of a multiresidue block in the FF,
+           all nodes corresponding to that block form a fragment.
+           All fragments are added to the fragments attribute, and
+           the nodes in those framgnets all have the entry node_to_block
+           set to the block. In addition it is recorded to which fragment
+           specifically the node belongs in the node_to_fragment dict.
+        3) the node corresponds to a multiresidue block; but unlike in
+           case to it represents multiple residues. ....
+
+         Parameters
+        ----------
+        meta_molecule: polyply.src.meta_molecule.MetaMolecule
+            The meta molecule to process.
+
         """
-        # 0. make edges for the block
-        for inter_type in ["bonds", "constraints", "virtual_sitesn",
-                           "virtual_sites2", "virtual_sites3", "virtual_sites4"]:
-            block.make_edges_from_interaction_type(inter_type)
-
-        # 1. make residue graph
-        expanded_graph = make_residue_graph(block)
-
-        # 2. clear out all old edges, because the
-        # inserted part will not have any edges with the
-        # rest of the meta_molecule as we don't know those
-        old_edges = list(meta_molecule.edges(meta_mol_node))
-        meta_molecule.remove_edges_from(old_edges)
-
-        # 3. relable nodes to make space for new nodes to be inserted
-        mapping = {}
-        offset = len(expanded_graph.nodes) - 1
-        for node in meta_molecule.nodes:
-            if node > meta_mol_node:
-                mapping[node] = node + offset
-
-        nx.relabel_nodes(meta_molecule, mapping, copy=False)
-        # 4. add the new nodes to the meta molecule overwriting
-        # the inital nodes inserting the graph
-        nodes = sorted(expanded_graph.nodes)
-        for node in nodes:
-            node_key = node + meta_mol_node
-            attrs = expanded_graph.nodes[node]
-            attrs["links"] = False
-            meta_molecule.add_node(node_key, **attrs)
-
-        # 5. add all edges within the exapnded block
-        for edge in expanded_graph.edges:
-            meta_molecule.add_edge(edge[0] + meta_mol_node, edge[1] + meta_mol_node)
+        regular_graph = nx.Graph()
+        restart_graph = nx.Graph()
+        restart_attr = nx.get_node_attributes(meta_molecule, "from_itp")
+
+        # this breaks down when to proteins are directly linked
+        # but that is an edge-case we can worry about later
```

Comment:
Why does that break?

Reply:
because they would appear as one connected component and not two seperate components referring to two molecules

I added this in the comment as well